dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -106,47 +94,53 @@ ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnD
     return columns_to_read[column_index];
 }
 
-enum class OperandType
-{
-    Unknown = 0,
-    Column,
-    Literal,
-};
-
 inline RSOperatorPtr parseTiCompareExpr( //
     const tipb::Expr & expr,
     const FilterParser::RSFilterType filter_type,
     const ColumnDefines & columns_to_read,
     const FilterParser::AttrCreatorByColumnID & creator,
-    const TimezoneInfo & timezone_info,
-    const LoggerPtr & /*log*/)
+    const TimezoneInfo & timezone_info)
 {
-    if (unlikely(
-            expr.children_size() != 2 && filter_type != FilterParser::RSFilterType::In
-            && filter_type != FilterParser::RSFilterType::NotIn))
+    if (unlikely(expr.children_size() != 2 && filter_type != FilterParser::RSFilterType::In))
         return createUnsupported(
             expr.ShortDebugString(),
-            tipb::ScalarFuncSig_Name(expr.sig()) + " with " + DB::toString(expr.children_size())
-                + " children is not supported",
+            fmt::format(
+                "{} with {} children is not supported",
+                tipb::ScalarFuncSig_Name(expr.sig()),
+                expr.children_size()),
             false);
 
-    /// Only support `column` `op` `literal`, and `column` in (literal1, literal2, ...) now.
+    // Support three types of expression:
+    // 1. `column` `op` `literal`
+    // 2. `literal` `op` `column`
+    // 3. `column` in (literal1, literal2, ...)
```

Comment:
```suggestion
    // 3. `column` in (literal1, literal2, ...); Note that the children size of `col in (1,2)` is 3
```

Reply:
https://github.com/pingcap/tiflash/pull/8212/commits/0bcc39e0ef48ff9d66c804cc5ba6661b4a22f062

change to 
```
    // 1. op(column, literal), in sql: column op literal
    // 2. op(literal, column), in sql: literal op column
    // 3. in(coloumn, literal1, literal2, ...), in sql: column in (literal1, literal2, ...)
    // op is one of: =, !=, >, >=, <, <=
```